internal/controller/etcdcluster_controller.go

```diff
@@ -47,17 +59,193 @@ type EtcdClusterReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.1/pkg/reconcile
 func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch the EtcdCluster resource
+	etcdCluster := &ecv1alpha1.EtcdCluster{}
+
+	err := r.Get(ctx, req.NamespacedName, etcdCluster)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("EtcdCluster resource not found. Ignoring since object may have been deleted")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
```

Comment:
let's add a check something like below: 
```
if etcdCluster.Size == 0 {
   print a warning
   return
}
```

Afterwards, simplify the following logic since we can assume that etcdCluster.Size is guaranteed to be > 0.